src/ViewHelpers/VariableViewHelper.php

```diff
@@ -0,0 +1,71 @@
+<?php
+namespace TYPO3Fluid\Fluid\ViewHelpers;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+use TYPO3Fluid\Fluid\Core\Compiler\TemplateCompiler;
+use TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\ViewHelperNode;
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper;
+use TYPO3Fluid\Fluid\Core\ViewHelper\Traits\CompileWithContentArgumentAndRenderStatic;
+
+/**
+ * Variable assigning ViewHelper
+ *
+ * Assigns one template variable which will exist also
+ * after the ViewHelper is done rendering, i.e. adds
+ * template variables.
+ *
+ * If you require a variable assignment which does not
+ * exist in the template after a piece of Fluid code
+ * is rendered, consider using `f:alias` instead.
+ *
+ * Usages:
+ *
+ *     {f:variable(name: 'myvariable', value: 'some value')}
+ *     <f:variable name="myvariable">some value</f:variable>
+ *     {oldvariable -> f:format.htmlspecialchars() -> f:variable(name: 'newvariable')}
+ *     <f:variable name="myvariable"><f:format.htmlspecialchars>{oldvariable}</f:format.htmlspecialchars></f:variable>
+ *
+ * Please note:
+ *
+ * If you assign a variable with a dotted name, e.g.
+ * "myarray.property" then this may or may not be
+ * supported by the VariableProvider that's being
+ * used. The default
```

Comment:
+ i'm not really sure we should allow a dot in the name. My gut tells me, that prohibiting dot here will save use plenty of headaches later on